### PR TITLE
update warning message

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*
#1582 

*Description of changes:*
raise warning for parameter server + multi gpu for `local_gpu`

*Testing done:*
unit test

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/fw_utils.py [hunk 1]
```diff
@@ -617,9 +617,9 @@ def warn_if_parameter_server_with_multi_gpu(training_instance_type, distribution
         return
 
     is_multi_gpu_instance = (
-        training_instance_type.split(".")[1].startswith("p")
-        and training_instance_type not in SINGLE_GPU_INSTANCE_TYPES
-    )
+        training_instance_type == "local_gpu"
```
Comment (ChaiBapchya) [inline src/sagemaker/fw_utils.py:620]: how are we checking if its a multi_gpu instance just by checking for "local_gpu"
or is it assumed that the person who is testing locally has verified that his local instance supports multi_gpu

File: tests/unit/test_fw_utils.py [hunk 2]
```diff
@@ -1272,3 +1272,13 @@ def test_warn_if_parameter_server_with_multi_gpu(caplog):
         training_instance_type=train_instance_type, distributions=distributions
     )
     assert fw_utils.PARAMETER_SERVER_MULTI_GPU_WARNING in caplog.text
+
+
+def test_war_if_parameter_server_with_multi_gpu(caplog):
```
Comment (ChaiBapchya) [inline tests/unit/test_fw_utils.py:1277]:
```suggestion
def test_warn_if_parameter_server_with_multi_gpu(caplog):
```

File: src/sagemaker/fw_utils.py (hunk 1 above)
Comment (ChuyangDeng) [reply src/sagemaker/fw_utils.py:620]: We cannot check that, so I updated the warning message to "if"

File: src/sagemaker/fw_utils.py [hunk 3]
```diff
@@ -617,9 +617,9 @@ def warn_if_parameter_server_with_multi_gpu(training_instance_type, distribution
...
+        or training_instance_type.split(".")[1].startswith("p")
+    ) and training_instance_type not in SINGLE_GPU_INSTANCE_TYPES
 
     ps_enabled = "parameter_server" in distributions and distributions["parameter_server"].get(
```
Comment (laurenyu) [inline src/sagemaker/fw_utils.py:624]: `"local_gpu"` doesn't support distributed training iirc, so if indeed a parameter server is set, wouldn't it make more sense to assume that the intent is to do distributed training and instead fail that GPU + distributed doesn't work without printing out this ambiguous warning that won't apply?

**Pull request merged**